st2client/st2client/models/core.py

```diff
@@ -274,11 +274,14 @@ def query(self, **kwargs):
             self.handle_error(response)
         items = response.json()
         instances = [self.resource.deserialize(item) for item in items]
-        return instances
+        if 'X-Total-Count' in response.headers:
```

Comment:
It looks like query is used in quite some places, were all the affected places updated?